src/components/shared/Drawer.jsx

```diff
@@ -0,0 +1,84 @@
+import Link from 'next/link';
+import { FaSearch } from 'react-icons/fa';
+import './Header.css';
+
+const Drawer = ({ setOpenDrawer, user, openDrawer }) => {
+  return (
+    <div
+      className={`slide ${openDrawer ? 'open' : ''} min-h-[100vh] w-[70vw]  overflow-hidden bg-california-50 p-4 text-sushi-700 lg:w-[30vw]`}
+    >
+      <div className='flex justify-end'>
+        <button onClick={() => setOpenDrawer(false)}>X</button>
+      </div>
+
+      {user && (
+        <div className='mb-4 flex items-center gap-3 border-b pb-4'>
+          <img
+            src='https://media.newyorker.com/photos/644ab20d87ae684cabca21f2/master/pass/knight-Erling-Haaland.jpg'
+            alt=''
+            className='h-12 w-12 rounded-full'
+          />
```

Comment:
We are using NextJS, here has an in-built <Image /> tag for showing images with optimization and other benefits. We need to use this tag for showing images in NextJS.

Reply:
I have used <Image> tag provided by Next.js. However, you'll notice I have used height & width as both Tailwind classes and as props as well. This is because, if 'height' & 'width' props are not used, an error pops up, but due to adjustment issues like height not working properly, I have still kept Tailwind classes 'h-10' & 'w-10'.